src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskCancelByIdControllerTest.java

```diff
@@ -0,0 +1,117 @@
+
+class PostTaskCancelByIdControllerTest extends SpringBootIntegrationBaseTest {
+
+    private static final String ENDPOINT_PATH = "/task/%s/cancel";
+    private static String ENDPOINT_BEING_TESTED;
+    @MockBean
+    private IdamWebApi idamWebApi;
+    @MockBean
+    private CamundaServiceApi camundaServiceApi;
+    @MockBean
+    private AuthTokenGenerator authTokenGenerator;
+    @MockBean
+    private RoleAssignmentServiceApi roleAssignmentServiceApi;
+    @MockBean
+    private ServiceAuthorisationApi serviceAuthorisationApi;
+    @MockBean
+    private AccessControlService accessControlService;
+    @MockBean
+    private PermissionEvaluatorService permissionEvaluatorService;
+    @Mock
+    private Assignment mockedRoleAssignment;
+    @Mock
+    private UserInfo mockedUserInfo;
+
+    private ServiceMocks mockServices;
+    private String taskId;
+
+    @BeforeEach
+    void setUp() {
+        taskId = UUID.randomUUID().toString();
+        ENDPOINT_BEING_TESTED = String.format(ENDPOINT_PATH, taskId);
+
+        when(authTokenGenerator.generate())
+            .thenReturn(IDAM_AUTHORIZATION_TOKEN);
+        when(mockedUserInfo.getUid())
+            .thenReturn(IDAM_USER_ID);
+        when(accessControlService.getRoles(IDAM_AUTHORIZATION_TOKEN))
+            .thenReturn(new AccessControlResponse(mockedUserInfo, singletonList(mockedRoleAssignment)));
+        mockServices = new ServiceMocks(
+            idamWebApi,
+            serviceAuthorisationApi,
+            camundaServiceApi,
+            roleAssignmentServiceApi);
+    }
+
+    @Test
+    void should_return_500_with_application_problem_response_when_cancel_call_fails() throws Exception {
+
+        mockServices.mockServiceAPIs();
+
+        when(permissionEvaluatorService.hasAccess(any(), any(), any()))
+            .thenReturn(true);
+
+        CamundaTask camundaTasks = mockServices.getCamundaTask("processInstanceId", taskId);
+        when(camundaServiceApi.getTask(any(), eq(taskId))).thenReturn(camundaTasks);
+
+        doThrow(FeignException.FeignServerException.class).when(camundaServiceApi).bpmnEscalation(any(), any(), any());
+
+        mockMvc.perform(
+            post(ENDPOINT_BEING_TESTED)
+                .header(AUTHORIZATION, IDAM_AUTHORIZATION_TOKEN)
+                .header(SERVICE_AUTHORIZATION, SERVICE_AUTHORIZATION_TOKEN)
+                .contentType(MediaType.APPLICATION_JSON_VALUE)
+        ).andExpect(
+            ResultMatcher.matchAll(
+                status().is5xxServerError(),
+                content().contentType(APPLICATION_PROBLEM_JSON_VALUE),
+                jsonPath("$.type").value("https://github.com/hmcts/wa-task-management-api/problem/task-cancel-error"),
+                jsonPath("$.title").value("Task Cancel Error"),
+                jsonPath("$.status").value(500),
+                jsonPath("$.detail").value(
+                    "Task Cancel Error: Unable to cancel the task.")
+            ));
+    }
```

Comment:
Just curious, Does a task need to be reassigned to be cancelled? I.e. can tasks be cancelled at any point by anyone who has permissions?